packages/text-area/src/vaadin-text-area.js

```diff
@@ -289,6 +291,29 @@ export class TextArea extends InputFieldMixin(ThemableMixin(ElementMixin(Polymer
 
     this._dispatchIronResizeEventIfNeeded('InputHeight', inputHeight);
   }
+
+  checkValidity() {
+    if (!super.checkValidity()) {
+      return false;
+    }
+
+    // Custom pattern validation implementation, according to web standards spec
+
+    // Skip pattern validation if there is no pattern, or the value is empty
+    if (!this.pattern || !this.inputElement.value) {
+      return true;
+    }
+
+    // Skip if pattern is invalid
+    try {
+      new RegExp(this.pattern);
```

Comment:
Isn't this check redundant because below `this.inputElement.value.match(this.pattern)` creates a RegExp instance as well? https://262.ecma-international.org/5.1/#sec-15.5.4.10

I would suggest to wrap the call below, instead of parsing the pattern twice.

Reply:
Done